src/components/TopicList/index.js

```diff
@@ -1,20 +1,13 @@
-import React, { useEffect } from 'react';
+import React from 'react';
 import { FlatList, View } from 'react-native';
-import { useDispatch, useSelector } from 'react-redux';
-import { getTopics } from 'actions/topicActions';
+import { useSelector } from 'react-redux';
 import TopicListItem from 'components/TopicListItem';
 import { func } from 'prop-types';
 import styles from './styles';
 
 const TopicList = ({ setModalVisible, onValueChange }) => {
-  const dispatch = useDispatch();
-
-  useEffect(() => {
-    dispatch(getTopics());
-  }, []);
-
   const { topics } = useSelector(({ topics }) => ({
-    topics: topics?.topics?.topics ?? [],
+    topics: topics?.topics ?? [],
```

Comment:
Yes, that's right. It's just a list that shows topics but I thought that it would be more readable and clean if the logic from selecting the topics from the state was in the component that need the info instead of everything mixed in the parent. Because in the parent component also fetch targets. Anyway, I understand what you're saying and I could work on it.